nginx-controller/nginx/nginx.go

```diff
@@ -48,8 +49,19 @@ type UpstreamServer struct {
 	Port    string
 }
 
+// Map represents map directive in NGINX config
+type Map struct {
+	Source   string
+	Target   string
+	Variable string
+	Values   map[string]string
+}
+
 // Server describes an NGINX server
 type Server struct {
+	Namespace   string
```

Comment:
Namespace, IngressName as well as Maps should be part of IngressConfig.

Reply:
done